raft: nil pointer dereference in multiNode.run

The following test appears to have failed:

[#8932](https://circleci.com/gh/cockroachdb/cockroach/8932):

```
I1105 00:35:09.234897 929 storage/replica_gc_queue.go:125  destroying local data from range 1
I1105 00:35:09.235841 929 storage/store.go:1475  changing raft replica {3 3 8} for range 1
E1105 00:35:09.237934 929 storage/store.go:1543  storage/store.go:1542: got committed raft command for 1 but have no range with that ID: {RangeID:1 OriginReplica:{NodeID:1 StoreID:1 ReplicaID:1} Cmd:BeginTransaction ["\x00\x00\x00k1\x00\x01rdsc",""), ConditionalPut ["\x00\x00\x00k1\x00\x01rdsc",""), Put ["\x00\x00meta2\xff\xff",""), Put ["\x00\x00meta1\xff\xff","")}
E1105 00:35:09.238132 929 storage/store.go:1543  storage/store.go:1542: got committed raft command for 1 but have no range with that ID: {RangeID:1 OriginReplica:{NodeID:1 StoreID:1 ReplicaID:1} Cmd:EndTransaction ["\x00\x00\x00k1\x00\x01rdsc","")}
W1105 00:35:09.239251 929 multiraft/multiraft.go:1151  aborting configuration change: storage/store.go:1542: got committed raft command for 1 but have no range with that ID: {RangeID:1 OriginReplica:{NodeID:1 StoreID:1 ReplicaID:1} Cmd:EndTransaction ["\x00\x00\x00k1\x00\x01rdsc","")}
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x8 pc=0xdc73ff]

goroutine 6856 [running]:
github.com/coreos/etcd/raft.(*multiNode).run(0xc820303c80)
	/go/src/github.com/coreos/etcd/raft/multinode.go:264 +0x174f
created by github.com/coreos/etcd/raft.StartMultiNode
	/go/src/github.com/coreos/etcd/raft/multinode.go:71 +0x338

goroutine 1 [chan receive]:
testing.RunTests(0x1891350, 0x1ea96a0, 0xc9, 0xc9, 0x1568201)
--
	/go/src/github.com/cockroachdb/cockroach/multiraft/storage.go:199 +0x1031
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820302840, 0xc82050fbc0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
FAIL	github.com/cockroachdb/cockroach/storage	19.019s
=== RUN   TestBatchBasics
I1105 00:34:48.318960 923 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I1105 00:34:48.323236 923 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I1105 00:34:48.329965 923 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

another one from `TestRaftRemoveRace` in https://circleci.com/gh/tschottdorf/cockroach/1686

```
I1120 02:16:16.902423 1013 storage/store.go:1506  raft: REMOVE_REPLICA {3 3 4} for range 1
W1120 02:16:16.903514 1013 multiraft/multiraft.go:1092  failed to look up replica ID for range 1 (disabling replica ID check): storage/store.go:1655: store 3 not found as replica of range 1
I1120 02:16:16.903827 1013 util/tracer/tracer.go:158  Name              Origin  Ts               Dur          Desc                     File
Rd@0.000000000,0          02:16:16.884567  18.978657ms  sending batch            kv/txn_coord_sender.go:309
Rd@0.000000000,0          02:16:16.884578  6.579µs      ·meta descriptor lookup  kv/dist_sender.go:515
Rd@0.000000000,0          02:16:16.884592  12.690643ms  ·sending RPC             kv/dist_sender.go:421

I1120 02:16:16.903875 1013 storage/replica_gc_queue.go:126  destroying local data from range 1
I1120 02:16:16.904951 1013 storage/replica_command.go:458  auto-gc'ed dbd06763 (3 intents)
E1120 02:16:16.905337 1013 storage/store.go:1574  storage/store.go:1573: got committed raft command for 1 but have no range with that ID: {RangeID:1 OriginReplica:{NodeID:1 StoreID:1 ReplicaID:1} Cmd:BeginTransaction [/Local/Range/RangeDescriptor/"",/Min), ConditionalPut [/Local/Range/RangeDescriptor/"",/Min), Put [/System/Meta2/"\xff\xff",/Min), Put [/System/Meta1/"\xff\xff",/Min)}
E1120 02:16:16.905423 1013 storage/store.go:1574  storage/store.go:1573: got committed raft command for 1 but have no range with that ID: {RangeID:1 OriginReplica:{NodeID:1 StoreID:1 ReplicaID:1} Cmd:PushTxn [/Local/Range/RangeDescriptor/"",/Min)}
E1120 02:16:16.905498 1013 storage/store.go:1574  storage/store.go:1573: got committed raft command for 1 but have no range with that ID: {RangeID:1 OriginReplica:{NodeID:1 StoreID:1 ReplicaID:1} Cmd:ResolveIntent [/System/Meta1/"\xff\xff",/Min)}
E1120 02:16:16.905590 1013 storage/store.go:1574  storage/store.go:1573: got committed raft command for 1 but have no range with that ID: {RangeID:1 OriginReplica:{NodeID:1 StoreID:1 ReplicaID:1} Cmd:EndTransaction [/Local/Range/RangeDescriptor/"",/Min)}
W1120 02:16:16.906176 1013 multiraft/multiraft.go:1175  aborting configuration change: storage/store.go:1573: got committed raft command for 1 but have no range with that ID: {RangeID:1 OriginReplica:{NodeID:1 StoreID:1 ReplicaID:1} Cmd:EndTransaction [/Local/Range/RangeDescriptor/"",/Min)}
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x8 pc=0xc782da]

goroutine 6338 [running]:
github.com/coreos/etcd/raft.(*multiNode).run(0xc82079b920)
	/go/src/github.com/coreos/etcd/raft/multinode.go:264 +0x129a
created by github.com/coreos/etcd/raft.StartMultiNode
	/go/src/github.com/coreos/etcd/raft/multinode.go:71 +0x30d
```

--- a/multiraft/multiraft.go
+++ b/multiraft/multiraft.go
@@ -1154,7 +1154,13 @@ func (s *state) processCommittedEntry(groupID roachpb.RangeID, g *group, entry r
 			Callback: func(err error) {
 				select {
 				case s.callbackChan <- func() {
-					if err == nil {
+					gInner, ok := s.groups[groupID]
+					if !ok {
+						log.Infof("group %d no longer exists, aborting configuration change", groupID)
+					} else if gInner != g {
+						panic(fmt.Sprintf("passed in group and fetched group objects do not match\noriginal:%+v\nfetched:%+v\n",
+							g, gInner))
+					} else if err == nil {
 						if log.V(3) {
 							log.Infof("node %v applying configuration change %v", s.nodeID, cc)
 						}

